python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -105,6 +110,10 @@ def validate(self):
         if not self.differencer.doWriteSubtractedExp:
             raise ValueError("Source association needs difference exposures "
                              "[differencer.doWriteSubtractedExp].")
+        if self.diaCatalogLoader.htmLevel != \
+           self.associator.diaCalculation.plugins["ap_HTMIndex"].htmLevel:
+            raise ValueError("HTM index level in LoadDiaCatalogsTask must be equal to "
+                             "HTMIndexDiaCalculationPlugin index level.")
```

Comment:
:+1: for more validation, but does this need a check for whether the plugin is activated?

Reply:
Not here, but certainly within `AssociationTask` as none of the loading or storing in the Apdb will work without it. Added the validation to `AssociationTask`